### PR TITLE
Introduce a new PodMonitor to the Prometheus operator

### PR DESCRIPTION
**Pull request by valiparsa**
### Description

Make scraped metrics visible to the Prometheus operator through this newly introduced PodMonitor object.

Using this PodMonitor the Prometheus operator will scrape the configured endpoints every 5 seconds interval. 5 second interval is chosen due to short-lived Jenkinsfile-runner pods.

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [ ] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [ ] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: charts/steward/templates/podmonitor-jenkinsfile-runner.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.metrics.podMonitors.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: log-forwarder
```
Comment (romanisb) [inline charts/steward/templates/podmonitor-jenkinsfile-runner.yaml:5]: Naming is a bit off between the resource name and the file name.

Comment (valiparsa) [reply charts/steward/templates/podmonitor-jenkinsfile-runner.yaml]: Renamed here https://github.com/SAP/stewardci-core/pull/410/commits/406df9317653d079d855b78241a5d1a1e80a2327

File: charts/steward/values.yaml [hunk 2]
```diff
@@ -43,6 +43,9 @@ metrics:
   serviceMonitors:
     enabled: false
     extraLabels: {}
+  podMonitors:
+    enabled: false
+    extraLabels: {}
```
[the remark this answers is not in the document]
Comment (valiparsa) [reply charts/steward/values.yaml]: The `metric` section is in the higher level and it is not dedicated to the steward-run-controller. I added two child keys for the `metrics` key in the values.yaml to differentiate between JFR podmonitor and run-controller servicemonitor. Please check if it makes more sense now or not?

[the remark this answers is not in the document]
Comment (valiparsa) [reply charts/steward/values.yaml]: see https://github.com/SAP/stewardci-core/pull/410/commits/406df9317653d079d855b78241a5d1a1e80a2327

File: charts/steward/templates/podmonitor-jenkinsfile-runner.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    {{- include "steward.runController.componentLabel" . | nindent 4 }}
+    {{- include "steward.podMonitors.extraLabels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "steward.jenkinsfileRunner.selectorLabels" . | nindent 6 }}
+  namespaceSelector:
+    any: true
+  podMetricsEndpoints:
+  - port: prometheus
```
[the remark this answers is not in the document]
Comment (valiparsa) [reply charts/steward/templates/podmonitor-jenkinsfile-runner.yaml]: I renamed to `http-metrics` now
https://github.com/SAP/stewardci-core/pull/410/commits/406df9317653d079d855b78241a5d1a1e80a2327

File: charts/steward/templates/podmonitor-jenkinsfile-runner.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    {{- include "steward.runController.componentLabel" . | nindent 4 }}
+    {{- include "steward.jenkinsfileRunnerLogForwarder.podMonitors.extraLabels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "steward.jenkinsfileRunner.selectorLabels" . | nindent 6 }}
+  namespaceSelector:
+    any: true
+  podMetricsEndpoints:
+  - port:  http-metrics
```
Comment (romanisb) [inline charts/steward/templates/podmonitor-jenkinsfile-runner.yaml:18]:
```suggestion
  - port: http-metrics
```

File: changelog.yaml [hunk 5]
```diff
@@ -56,6 +56,19 @@
   date: 2023-10-06
   changes:
 
+    - type: enhancement
+      impact: minor
```
Comment (romanisb) [inline changelog.yaml:60]: The versions in the [Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) file needs to be adapted accordingly.

Comment (valiparsa) [reply changelog.yaml:60]: Done in https://github.com/SAP/stewardci-core/pull/410/commits/ea3403ac1a8e2696997cbc5838232529e5b11e5c

**Review by jaythamke — APPROVED** [review]
LGTM

**Comment by valiparsa** [comment]
After a discussion with @anbrsap we decided to do this differently. Hence, I'll close this PR and open new one.

**Pull request closed without merging**